libshaderc/src/shaderc_test.cc

```diff
@@ -323,6 +338,19 @@ TEST(CompileStringWithOptions, GlobalWarnings) {
               HasSubstr("version 550 is unknown.\n"));
 }
 
+TEST(CompileStringWithOptions, GlobalWarningsAsErrors) {
+  Compiler compiler;
+  compile_options_ptr options(shaderc_compile_options_initialize());
+  shaderc_compile_options_set_warnings_as_errors(options.get());
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const Compilation comp(compiler.get_compiler_handle(),
+                         kMinimalUnknownVersionShader,
+                         shaderc_glsl_vertex_shader, options.get());
+  EXPECT_FALSE(shaderc_module_get_success(comp.result()));
+  EXPECT_THAT(shaderc_module_get_error_message(comp.result()),
+              HasSubstr("error: version 550 is unknown.\n"));
+}
+
 TEST(CompileStringWithOptions, SuppressGlobalWarnings) {
```

Comment:
The suppression cases are included in SuppressWarningsMode<First/Second>OverridesWarningsAsErrorsMode.
The cases with Unknown version shader.